numpy/distutils/mingw32ccompiler.py

```diff
@@ -124,10 +135,10 @@ def __init__ (self,
             else:
                 # gcc-4 series releases do not support -mno-cygwin option
                 self.set_executables(
-                    compiler='gcc -g -DDEBUG -DMS_WIN64 -O0 -Wall',
-                    compiler_so='gcc -g -DDEBUG -DMS_WIN64 -O0 -Wall -Wstrict-prototypes',
-                    linker_exe='gcc -g',
-                    linker_so='gcc -g -shared')
+                    compiler='gcc -march=x86-64 -mtune=generic -DMS_WIN64 -O2 -msse2 -Wall',
+                    compiler_so='gcc -march=x86-64 -mtune=generic -DMS_WIN64 -O2 -msse2 -Wall -Wstrict-prototypes',
```

Comment:
After numerous tests with different combinations of gcc flags I came to the conclusion, that the settings for the gcc flags in the PR seems to be almost correct. `-march=x86-64 -mtune=generic` is recommended for amd64 usage. `-march=core2 -mtune=generic` could be changed to `-march=pentium4 -mtune=generic`.